pkg/component/kubernetes/apiserver/authentication.go

```diff
@@ -44,90 +45,113 @@ func (k *kubeAPIServer) reconcileConfigMapAuthenticationConfig(ctx context.Conte
+	jwts[0].ClaimMappings.Username.Claim = ptr.Deref(oidc.UsernameClaim, "sub")
 
 	usernamePrefix := ptr.Deref(oidc.UsernamePrefix, "")
 
 	// For the authentication config, there is no defaulting for claim or prefix.
 	// We use the defaulting suggested in kubernetes https://github.com/kubernetes/kubernetes/blob/a2106b5f73fe9352f7bc0520788855d57fc301e1/staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go#L357-L366
 	switch {
 	case usernamePrefix == "-":
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To("")
-	case len(usernamePrefix) == 0 && authenticationConfiguration.JWT[0].ClaimMappings.Username.Claim != "email":
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To(fmt.Sprintf("%s#", authenticationConfiguration.JWT[0].Issuer.URL))
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To("")
+	case len(usernamePrefix) == 0 && jwts[0].ClaimMappings.Username.Claim != "email":
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To(fmt.Sprintf("%s#", jwts[0].Issuer.URL))
 	default:
-		authenticationConfiguration.JWT[0].ClaimMappings.Username.Prefix = ptr.To(usernamePrefix)
+		jwts[0].ClaimMappings.Username.Prefix = ptr.To(usernamePrefix)
 	}
 
 	if v := oidc.GroupsClaim; v != nil {
-		authenticationConfiguration.JWT[0].ClaimMappings.Groups.Claim = *v
-		authenticationConfiguration.JWT[0].ClaimMappings.Groups.Prefix = ptr.To(ptr.Deref(oidc.GroupsPrefix, ""))
+		jwts[0].ClaimMappings.Groups.Claim = *v
+		jwts[0].ClaimMappings.Groups.Prefix = ptr.To(ptr.Deref(oidc.GroupsPrefix, ""))
 	}
 
 	for key, value := range oidc.RequiredClaims {
 		claimValidationRule := apiserverv1beta1.ClaimValidationRule{
 			Claim:         key,
 			RequiredValue: value,
 		}
-		authenticationConfiguration.JWT[0].ClaimValidationRules = append(authenticationConfiguration.JWT[0].ClaimValidationRules, claimValidationRule)
+		jwts[0].ClaimValidationRules = append(jwts[0].ClaimValidationRules, claimValidationRule)
 	}
 
-	data, err := runtime.Encode(ConfigCodec, authenticationConfiguration)
-	if err != nil {
-		return "", fmt.Errorf("unable to encode authentication configuration: %w", err)
-	}
-
-	return string(data), nil
+	return jwts
 }
 
 func (k *kubeAPIServer) handleAuthenticationSettings(deployment *appsv1.Deployment, configMapAuthenticationConfig *corev1.ConfigMap, secretOIDCCABundle *corev1.Secret) {
 	if value, ok := k.values.FeatureGates["StructuredAuthenticationConfiguration"]; versionutils.ConstraintK8sLess130.Check(k.values.Version) || (ok && !value) {
 		k.handleOIDCSettings(deployment, secretOIDCCABundle)
-		return
+	}
+
+	var (
+		anonymousGateEnabled, anonymousGateSet    = k.values.FeatureGates["AnonymousAuthConfigurableEndpoints"]
+		shouldConfigureAnonymousViaStructuredAuth = (!anonymousGateSet || (anonymousGateSet && anonymousGateEnabled)) && versionutils.ConstraintK8sGreaterEqual132.Check(k.values.Version)
+	)
+
+	if !shouldConfigureAnonymousViaStructuredAuth {
```

Comment:
```suggestion
	if !shouldConfigureAnonymousViaStructuredAuth && !structAuthGateEnabled {
```

Should the `StructuredAuthenticationConfiguration` be checked here as well?

Reply:
Thanks for reviewing @timuthy. Those are some tricky statements. Should be addressed with https://github.com/gardener/gardener/pull/12198/commits/b5ef554ac615e8fc52fcbd259ec0b3a72eba2ff2

Mind that my additional check looks a bit different, but I think it should be correct